legend-engine-xts-dataquality/legend-engine-xt-dataquality-protocol/src/main/java/org/finos/legend/engine/protocol/dataquality/model/DataQualityExecuteInput.java

```diff
@@ -0,0 +1,30 @@
+//  Copyright 2022 Goldman Sachs
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package org.finos.legend.engine.protocol.dataquality.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.finos.legend.engine.protocol.pure.v1.model.context.AlloySDLC;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain.ParameterValue;
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DataQualityExecuteInput
+{
+    public AlloySDLC alloySDLC;
```

Comment:
Assuming this represents the model in the DQ Input. 
Does it make sense for this to be `PureModelContext` instead so we can run these checks both with grammar inputs as well as a full pmcd and not just pointed to alloy projects ?

Reply:
hi Mauricio, we do have an API which takes in a full pmcd, generates and execute the plan from it to pull defects. The request class is [DataQualityExecuteTrialInput](https://github.com/goldmansachs/legend-engine/blob/review-new-data-quality-module/legend-engine-xts-dataquality/legend-engine-xt-dataquality-api/src/main/java/org/finos/legend/engine/language/dataquality/api/DataQualityExecuteTrialInput.java) We will rename these classes better to DataQualityExecuteArtifactsInput and DataQualityExecuteInput. 

1. /execute - this API will be integrated in the studio where user can run the validation and fetch defects 
2. /executeArtifacts - this API will take in a reference to a project and then execute the stored plan to pull defects.(the above mentioned class in an input to this API)